roxie/ccd/ccdqueue.cpp

```diff
@@ -1769,23 +1998,26 @@ class RoxieSocketQueueManager : public RoxieReceiverBase
                 StringBuffer s;
                 throw MakeStringException(ROXIE_PACKET_ERROR, "Maximum packet length %d exceeded sending packet %s", maxPacketSize, header.toString(s).str());
             }
-            if (channelWrite(header.channel, &header, length) != length)
-                logctx.CTXLOG("multicast write wrote too little");
+            if (!channelWrite(header, true))
+                logctx.CTXLOG("Roxie packet write wrote too little");
             packetsSent++;
         }
     }
 
     virtual void sendIbyti(RoxiePacketHeader &header, const IRoxieContextLogger &logctx, unsigned subChannel) override
     {
+#ifdef SUBCHANNELS_IN_HEADER
+        if (!header.hasBuddies())
+            return;
+#endif
         MTIME_SECTION(queryActiveTimer(), "RoxieSocketQueueManager::sendIbyti");
         RoxiePacketHeader ibytiHeader(header, header.activityId & ROXIE_PRIORITY_MASK, subChannel);
     
         if (logctx.queryTraceLevel() > 8)
         {
             StringBuffer s; logctx.CTXLOG("Sending IBYTI packet %s", ibytiHeader.toString(s).str());
         }
-        if (channelWrite(header.channel, &ibytiHeader, sizeof(RoxiePacketHeader)) != sizeof(RoxiePacketHeader))
-            logctx.CTXLOG("sendIbyti wrote too little");
+        channelWrite(ibytiHeader, false);  // don't send to self
```

Comment:
Was loss of logging intentional?  Coverity may complain if not ll calls check the result.

Reply:
Yes it was intentional, as it's ok to write nothing if there are no buddy nodes.